pkg/services/ngalert/state/state.go

```diff
@@ -87,13 +89,26 @@ func (a *State) resultAlerting(alertRule *ngModels.AlertRule, result eval.Result
 
 func (a *State) resultError(alertRule *ngModels.AlertRule, result eval.Result) {
 	a.Error = result.Error
+
 	if a.StartsAt.IsZero() {
 		a.StartsAt = result.EvaluatedAt
 	}
 	a.setEndsAt(alertRule, result)
 
 	if alertRule.ExecErrState == ngModels.AlertingErrState {
 		a.State = eval.Alerting
+	} else if alertRule.ExecErrState == ngModels.ErrorErrState {
```

Comment:
I'm a bit confused about this part - I thought this alert would basically be attached to the above condition. When we say "alert me if this goes into an error".

```golang
alertRule.ExecErrState == ngModels.AlertingErrState
```

Reply:
Discussed this offline - I'm good with this change. Even if it adds more options which is a non-goal long term, at least is consistent.